Modules/ThirdParty/HDF5/src/itkhdf5/src/H5win32defs.h

```diff
@@ -150,6 +158,7 @@ extern "C" {
 #define HDpthread_self_ulong() ((unsigned long)GetCurrentThreadId())
 
 #ifndef H5_HAVE_MINGW
```

Comment:
We have `#ifndef H5_HAVE_MINGW`-- maybe mistaken for `#ifdef H5_HAVE_MINGW`?

What if we move this line to line 108,

```
#else
// For MinGW, etc.
#define HDopen(S,F,M)       _open(S, F | _O_BINARY, M)
#endif /* H5_HAVE_VISUAL_STUDIO */
```

?

Reply:
Pushing this...